src/els_completion_provider.erl

```diff
@@ -140,63 +158,59 @@ item_kind_module(Module) ->
    }.
 
 %%==============================================================================
-%% Functions
+%% Functions, Types, Macros and Records
 %%==============================================================================
 
--spec functions(els_dt_document:item(), boolean(), boolean()) -> [map()].
-functions(Document, _OnlyExported = false, Arity) ->
-  POIs = els_dt_document:pois(Document, [function]),
-  List = [completion_item_function(POI, Arity) || POI <- POIs],
-  lists:usort(List);
-functions(Document, _OnlyExported = true, Arity) ->
-  Exports   = els_dt_document:pois(Document, [export_entry]),
-  Functions = els_dt_document:pois(Document, [function]),
-  ExportsFA = [FA || #{id := FA} <- Exports],
-  List      = [ completion_item_function(POI, Arity)
-                || #{id := FA} = POI <- Functions, lists:member(FA, ExportsFA)
-              ],
-  lists:usort(List).
-
--spec completion_item_function(poi(), boolean()) -> map().
-completion_item_function(#{id := {F, A}, data := ArgsNames}, false) ->
-  #{ label            => list_to_binary(io_lib:format("~p/~p", [F, A]))
-   , kind             => ?COMPLETION_ITEM_KIND_FUNCTION
-   , insertText       => snippet_function_call(F, ArgsNames)
-   , insertTextFormat => ?INSERT_TEXT_FORMAT_SNIPPET
-   };
-completion_item_function(#{id := {F, A}}, true) ->
-  #{ label            => list_to_binary(io_lib:format("~p/~p", [F, A]))
-   , kind             => ?COMPLETION_ITEM_KIND_FUNCTION
-   , insertTextFormat => ?INSERT_TEXT_FORMAT_PLAIN_TEXT
-   }.
-
--spec exported_functions(module(), boolean()) -> [map()] | null.
-exported_functions(Module, Arity) ->
+-spec definitions(els_dt_document:item(), poi_kind()) -> [map()].
+definitions(Document, POIKind) ->
+  definitions(Document, POIKind, _ExportFormat = false, _ExportedOnly = false).
+
+-spec definitions(els_dt_document:item(), poi_kind(), boolean()) -> [map()].
+definitions(Document, POIKind, ExportFormat) ->
+  definitions(Document, POIKind, ExportFormat, _ExportedOnly = false).
+
+-spec definitions(els_dt_document:item(), poi_kind(), boolean(), boolean()) ->
+  [map()].
+definitions(Document, POIKind, ExportFormat, ExportedOnly) ->
+  POIs     = local_and_included_pois(Document, POIKind),
```

Comment:
Nit: weird indentation. Also, I suggest we drop the convention of aligning matching operators in function bodies and to limit its use when column-aligning record definitions, types and such. WDYT?

Reply:
I like the suggestion. 😄